### PR TITLE
Feature/spring boot 3.0

### PR DESCRIPTION
**Pull request by southeo**
Update to spring 3.1 and added actuator
[Jira](https://naturalis.atlassian.net/browse/DD-503?atlOrigin=eyJpIjoiZmVjMGY4Y2UzZGUwNGE0NjkyZDExYmU4ZjNkYjA2MzMiLCJwIjoiaiJ9)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -98,18 +103,30 @@
...
-      <artifactId>keycloak-spring-boot-starter</artifactId>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-starter-security</artifactId>
     </dependency>
+
+    <!-- For some reason, the program won't start without this -->
     <dependency>
       <groupId>org.springframework.boot</groupId>
-      <artifactId>spring-boot-starter-security</artifactId>
+      <artifactId>spring-boot-starter-tomcat</artifactId>
```
Comment (samleeflang) [inline pom.xml:128]: Probably it is lacking a webserver to run to application on. Another option would be to add spring-web to the pom. I think it is also baked in that lib

File: src/test/java/eu/dissco/core/digitalspecimenprocessor/repository/DigitalSpecimenRepositoryIT.java [hunk 2]
```diff
@@ -25,6 +29,8 @@ class DigitalSpecimenRepositoryIT extends BaseRepositoryIT {
 
   private DigitalSpecimenRepository repository;
 
+  private MockedStatic<Clock> clockMock;
```
Comment (samleeflang) [inline src/test/java/eu/dissco/core/digitalspecimenprocessor/repository/DigitalSpecimenRepositoryIT.java:32]: If we use .isAfter then we can remove the MockedStatic completly right? Or is still used somewhere

**Review by samleeflang — COMMENTED** [review]
Small comment and question

File: src/test/java/eu/dissco/core/digitalspecimenprocessor/repository/DigitalSpecimenRepositoryIT.java (hunk 2 above)
Comment (southeo) [reply src/test/java/eu/dissco/core/digitalspecimenprocessor/repository/DigitalSpecimenRepositoryIT.java:32]: Removed

File: pom.xml (hunk 1 above)
Comment (southeo) [reply pom.xml]: Changed to spring web, it works now

**Review by samleeflang — APPROVED** [review]

**Pull request merged**